oracle/oracle.go

```diff
@@ -0,0 +1,128 @@
+package oracle
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/republicprotocol/republic-go/crypto"
+	"github.com/republicprotocol/republic-go/dispatch"
+	"github.com/republicprotocol/republic-go/identity"
+	"github.com/republicprotocol/republic-go/logger"
+	"github.com/republicprotocol/republic-go/swarm"
+)
+
+// A Client exposes methods for invoking RPCs on a remote server.
+type Client interface {
+	// UpdateMidpoint is used to send updated midpoint information to a given
+	// multiAddress.
+	UpdateMidpoint(ctx context.Context, to identity.MultiAddress, midpointPrice MidpointPrice) error
+
+	// MultiAddress is used when finding random nodes to send information to.
+	MultiAddress() identity.MultiAddress
+}
+
+type Oracler interface {
+	// UpdateMidpoint sends the given midpoint information to α randomly
+	// selected nodes.
+	UpdateMidpoint(ctx context.Context, midpointPrice MidpointPrice) error
+
+	// MultiAddress is used when finding random nodes to send information to.
+	MultiAddress() identity.MultiAddress
```

Comment:
I feel that this doesn't have to be exposed by the `Oracler`, since the multi-address of the `Oracler` is not required by anyone outside the `Oracler`.